storage/s3.go

```diff
@@ -294,6 +294,30 @@ func (s *S3) listObjects(ctx context.Context, url *url.URL) <-chan *Object {
 	return objCh
 }
 
+// setAcl sets value of acl flag pointer according to the metadata
+// provided.
+func setAcl(acl **string, metadata map[string]string) error {
+	if acl == nil {
+		return fmt.Errorf("acl reference cannot point to nil")
+	}
+	aclVal := metadata["acl"]
+	if aclVal == "" {
+		return nil
+	}
+
+	supportedActions := map[string]bool{
```

Comment:
s5cmd supports other S3-compatible storage services. These "supported" actions may not be available, or there could be more options in other services. Let's keep the code simple and delegate the validation to the service.